torch/_inductor/device_properties.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any, Dict
+
+import torch
+from torch._dynamo.runtime import get_registered_device_runtimes, get_runtime_for_device
+
+# API to query cuda properties that will work in a triton compile process
+# that cannot use the GPU APIs (due to processing fork() and initialization
+# time issues). Properties are recorded in the main process before
+# we fork the workers.
+
+_compile_worker_device_properties: Dict[str, Any] = {}
+_compile_worker_current_devices: Dict[str, int] = {}
```

Comment:
This seems pretty similar to RuntimeInterface in runtime.py above.  Can we just merge both files into a single subclass?

Reply:
I have a try. But I met an issue. In my understanding, `DeviceInterface` is an interface of device runtime. So the API `set_device` and `get_device_properties` should be general. `set_compiler_worker_current_device`, `current_device`, and `get_device_properties` in device_properties.py are just used for compiling triton kernel in multi-processing.
Actually, `get_device_properties` in device_properties.py has a potential issue if the user uses `set_device` instead of `set_compiler_worker_current_device` to switch the device index. The reason is `current_device` in device_properties.py always fetch the cache from `_compile_worker_current_devices` instead of calling torch.cuda.current_device() in runtime.

Reply:
So I would like to know if it is reasonable to keep device_properties.py. If you prefer to merge them into `DeviceInterface`, is it ok to add two specific methods `set_device_for_cache` and `get_device_properties_for_cache` to 'DeviceInterface` to distinguish from `set_device` and `get_device_properties`.

Reply:
I think it is better to merge them unless there is some technical reason not to.

Reply:
done.